### PR TITLE
feat: Adds more options for s3 store

### PR DESCRIPTION
**Pull request by isaacsanders**
@asummers and I put this together to support CephFS.

### CODE REVIEW

**Comment by asummers** [comment]
The intuition here is that when using a non-Amazon S3, there's extra opts needed to propagate down to the `ExAWS.request` calls.

File: lib/mini_repo/store/s3.ex [hunk 1]
```diff
@@ -19,7 +19,9 @@ defmodule MiniRepo.Store.S3 do
       store =
         {MiniRepo.Store.S3,
          bucket: System.fetch_env!("MINI_REPO_S3_BUCKET"),
-         region: System.fetch_env!("MINI_REPO_S3_REGION")}
+         options: [
+          region: System.fetch_env!("MINI_REPO_S3_REGION")]
```
Comment (wojtekmach) [inline lib/mini_repo/store/s3.ex]:
```suggestion
           region: System.fetch_env!("MINI_REPO_S3_REGION")]
```

File: lib/mini_repo/store/s3.ex [hunk 2]
```diff
@@ -5,7 +5,7 @@ defmodule MiniRepo.Store.S3 do
   ## Options
 
     * `:bucket` - the S3 bucket
-    * `:region` - the S3 region
+    * `:options` - the S3 request options
```
Comment (wojtekmach) [inline lib/mini_repo/store/s3.ex]: Could you add a link to documentation with all possible options?

**Review by wojtekmach — COMMENTED** [review]
Good call, looks good. Just minor comments below.

**Comment by wojtekmach** [comment]
Thanks!

**Pull request merged**